configure.ac

```diff
@@ -379,61 +379,270 @@ fi
+   fi
+
+   if test "x$htop_curses_status$enable_unicode" = x0yes; then
+      AC_MSG_CHECKING([for mvadd_wchnstr in $htop_msg_linker_flags])
+      AC_LINK_IFELSE([AC_LANG_PROGRAM([[
+/* int mvadd_wchnstr(int y, int x, const cchar_t* wchstr, int n); */
+int mvadd_wchnstr(int y, int x, const void* wchstr, int n);
+         ]], [[
+mvadd_wchnstr(0, 0, (void*)0, 0);
+         ]])],
+         [AC_MSG_RESULT(yes)
+         htop_curses_capability=wide],
+         [AC_MSG_RESULT(no)
+         htop_curses_status=1])
+   fi
+
+   if test "$htop_curses_status" -eq 0; then
+      AM_CFLAGS="$AM_CFLAGS $htop_curses_cflags"
+      if test "$htop_curses_capability" = wide; then
+         AC_DEFINE([HAVE_LIBNCURSESW], 1, [libncursesw is present])
+      else
+         AC_DEFINE([HAVE_LIBNCURSES], 1, [libcurses is present])
+      fi
+   else
+      LIBS=$htop_save_LIBS
+   fi
+   CFLAGS=$htop_save_CFLAGS
+   return "$htop_curses_status"
+} # htop_check_curses_capability
+
+htop_curses_capability=none
+
+if test "x${CURSES_CFLAGS+y}${CURSES_LIBS+y}" = xyy; then
+   curses_pkg_names=""
+   htop_check_curses_capability "$CURSES_CFLAGS" "$CURSES_LIBS"
+fi
+
+# Prioritize $PKG_CONFIG over ncurses*-config, as users might need to
+# cross-compile htop.
+if test "x$PKG_CONFIG" != x; then
+   pkg_config_static_flag=""
+   if test "$enable_static" = yes; then
+      pkg_config_static_flag=--static
+   fi
+
+   for curses_name in $curses_pkg_names; do
+      echo retrieving $curses_name information through $PKG_CONFIG >&AS_MESSAGE_LOG_FD
+      $PKG_CONFIG --exists --print-errors $curses_name >&AS_MESSAGE_LOG_FD 2>&AS_MESSAGE_LOG_FD || continue
+
+      : # Resets "$?" to 0
+      htop_config_cflags=`$PKG_CONFIG --cflags $pkg_config_static_flag $curses_name 2>/dev/null` || continue
+      : # Resets "$?" to 0
+      htop_config_libs=`$PKG_CONFIG --libs $pkg_config_static_flag $curses_name 2>/dev/null` || continue
+
+      AC_MSG_RESULT([found $curses_name information through $PKG_CONFIG])
+
+      if htop_check_curses_capability "$htop_config_cflags" "$htop_config_libs"; then
+         break
+      fi
+   done
+fi
+
+case ${htop_curses_capability}-$enable_unicode in
+none-*|nonwide-yes)
+   for curses_name in $curses_pkg_names; do
+      echo retrieving $curses_name information through ${curses_name}-config >&AS_MESSAGE_LOG_FD
+
+      ${curses_name}-config --cflags >/dev/null 2>&AS_MESSAGE_LOG_FD || continue
+
+      : # Resets "$?" to 0
+      htop_config_cflags=`${curses_name}-config --cflags 2>/dev/null` || continue
+      : # Resets "$?" to 0
+      htop_config_libs=`${curses_name}-config --libs 2>/dev/null` || continue
+
+      AC_MSG_RESULT([found $curses_name information through ${curses_name}-config])
+
+      if htop_check_curses_capability "$htop_config_cflags" "$htop_config_libs"; then
+         break
+      fi
+   done
+   ;;
+esac
 
+case ${htop_curses_capability}-$enable_unicode in
+none-*|nonwide-yes)
+   if test "x$PKG_CONFIG" = x; then (
```

Comment:
Replying the [suggestion](https://github.com/htop-dev/htop/pull/1512#issuecomment-2359445533) from @BenBE.

The parentheses on that `if` block was intentional by me. This way I can isolate the local variables used within since all this block does is printing warning messages and it can all be done in a subshell.

Reply:
Reasonable. The way that scope is introduced just looks kinda strange without any annotation.

I'm also not sure if starting a subshell just for isolation is the right choice here or if  braces (`{}`) might do the trick too …

Reply:
The `{}` won't isolate local variables like I had originally intended.
If it was braces `{}` block, then it's no different from having no braces at all.

I wonder if I can make this code more obvious by turning into a shell function.